app/controllers/users_controller.rb

```diff
@@ -130,22 +132,34 @@ def full_delete
     end
   end
 
-  def remove_association
-    provider = params[:provider]
-    identity = @user.identities.find_by(provider: provider)
+  def remove_identity
     set_tabs("account")
 
-    if @user.identities.size == 2 && identity
+    error_message = "An error occurred. Please try again or send an email to: #{SiteConfig.email_addresses[:default]}"
+    unless Authentication::Providers.enabled?(params[:provider])
+      flash[:error] = error_message
+      redirect_to user_settings_path(@tab)
+      return
+    end
+
+    provider = Authentication::Providers.get!(params[:provider])
+
+    identity = @user.identities.find_by(provider: provider.provider_name)
+
+    if identity && @user.identities.size > 1
       identity.destroy
 
-      identity_username = "#{provider}_username".to_sym
-      @user.update(identity_username => nil, :profile_updated_at => Time.current)
+      @user.update(
+        provider.user_username_field => nil,
```

Comment:
Should this be `provider.class.user_username_field` like it is [below in the authenticator class](https://github.com/thepracticaldev/dev.to/pull/7597/files#diff-863cabff965798c30566df1bb5e2b3a8R99)?

Reply:
it's a bit scary that it didn't fail any test 👀 
Going to see why

Reply:
Oh I know why, that's the class object, not the instance object, that's why it works :D I'm going to apply @citizen428 suggestion here https://github.com/thepracticaldev/dev.to/pull/7597#discussion_r417809095 so we don't get confused ;-)